identity_iota_core/src/rebased/client/full_client.rs

```diff
@@ -253,18 +259,49 @@ impl<S> IdentityClient<S> {
     Ok(budget)
   }
 
-  async fn get_coin_for_transaction(&self) -> Result<iota_sdk::rpc_types::Coin, Error> {
-    let coins = self
-      .coin_read_api()
-      .get_coins(self.sender_address(), None, None, None)
-      .await
-      .map_err(|err| Error::GasIssue(format!("could not get coins; {err}")))?;
+  async fn get_coin_for_transaction(&self) -> Result<Coin, Error> {
+    const LIMIT: usize = 10;
+    let mut cursor = None;
+
+    let mut best_coin: Option<Coin> = None;
+
+    loop {
+      let coins = self
+        .coin_read_api()
+        .get_coins(self.sender_address(), None, cursor, Some(LIMIT))
+        .await?;
+
+      if coins.data.is_empty() {
+        return if let Some(coin) = best_coin {
+          Ok(coin)
+        } else {
+          Err(Error::GasIssue("no coins found for address".to_string()))
+        };
+      }
+
+      let page_best_coin = coins.data.into_iter().max_by_key(|coin| coin.balance);
+
+      if let Some(coin) = page_best_coin {
+        if coin.balance >= MINIMUM_BALANCE {
+          return Ok(coin);
+        }
+
+        if best_coin.as_ref().map_or(true, |best| coin.balance > best.balance) {
+          best_coin = Some(coin);
+        }
+      }
+
+      if !coins.has_next_page {
+        break;
+      }
+
+      cursor = coins.next_cursor;
+    }
 
-    coins
-      .data
-      .into_iter()
-      .next()
-      .ok_or_else(|| Error::GasIssue("could not find coins".to_string()))
+    Err(Error::GasIssue(format!(
+      "no coin found with minimum required balance of {}",
```

Comment:
Could you add the address to the error message? This might be helpful in case multiple accounts are used. ^^

Reply:
And the address? :D 

Same for error message in line 273 ("no coins found for address"). ^^